modules: mcuboot: Add Kconfig option for BOOT_MAX_ALIGN

pdgendt opened the pull request:
Add a configuration option to override the default BOOT_MAX_ALIGN value in `mcuboot`.

This can be used when image slots are on different flash devices.

pdgendt committed:
modules: mcuboot: Add Kconfig option for BOOT_MAX_ALIGN

Add a configuration option to override the default BOOT_MAX_ALIGN value in
mcuboot. It defaults to the chosen flash's write block size to make it
backwards compatible.

This can be used when image slots are on different flash devices.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

nordicjm commented on modules/Kconfig.mcuboot:
```diff
@@ -430,4 +430,14 @@ config MCUBOOT_BOOTUTIL_LIB_FOR_DIRECT_XIP
 
+DT_CHOSEN_ZEPHYR_FLASH := zephyr,flash
+
+config MCUBOOT_BOOT_MAX_ALIGN
```
this seems a bit problematic, the MCUboot code checks if this is defined - it will always be defined because there is no prompt or anything here, now what happens when a device is uses which does not have a write block size on this, or doesn't have a chosen flash node, does this work or through a CMake/Kconfig configuration error?

pdgendt commented on modules/Kconfig.mcuboot:
```diff
@@ -430,4 +430,14 @@ config MCUBOOT_BOOTUTIL_LIB_FOR_DIRECT_XIP
+	default $(dt_node_int_prop_int,$(DT_CHOSEN_ZEPHYR_FLASH),write-block-size)
+	help
+	  Allow to override the programmable flash block alignment size.
+	  By default it's set to the maximum of the write block size of
+	  the chosen zephyr,flash node and 8.
```
The value has to be the same for the bootloader, as for the application that will read/write/update the images (using the bootutil library). I found it difficult to pinpoint where this belong, but this seems to work.

nordicjm reviewed (commented):
When this is ready for update i.e. using proper commit, flip order of commits so Kconfig is added first, and let me know as I will prepare the MCUboot update commit

nordicjm committed:
west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  d5b0dcb9aaee397fc105ae2228e8030038c3d871

Brings following Zephyr relevant fixes:

  - d5b0dcb9 hooks: Use dedicated type for slot numbers
  - cf138a0f boot: zephyr: config: Introduce MCUBOOT_BOOT_MAX_ALIGN
    Kconfig symbol
  - 677f1602 boot: zephyr: boards: delete frdm_mcxa156 configuration
  - 3c8e5717 zephyr: nRF54L05 and L10 configuration with LTO enabled
  - 4ea993ce zephyr: Remove duplicate zephyr module line
  - 8d14eebf boot: bootutil: Fix upgrade only serial recovery slot
    info
  - 78f87cf3 boot: zephyr: boards config of the variant stm32 disco
    kit
  - cbd3310e boot: zephyr: get base addres from dts for flexspi
    devices
  - 425e22be boot/zephyr: fix compilation errors in logging.
  - ec26273f bootutil: fix null pointer dereference in logging.
  - 3eb23a5d zephyr: Kconfig to control MCUBOOT_USE_TLV_ALLOW_LIST
  - a13624f0 bootutil: Add MCUBOOT_USE_TLV_ALLOW_LIST
  - e375252c boot: Enable Encryption with PSA + ECDSA
  - 21ed226f boot: bootutil: Fix bootutil_find_key definition
  - 763be6a5 imgtool: Add sha512 to allowed configurations for
    ECDSA256P1
  - 59d2f7a5 imgtool: Add support for VID and CID
  - 94ad4d44 boot: Add VID and CID checks

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
